blueprints/blueprint/files/blueprints/__name__/index.js

```diff
@@ -1,25 +1,36 @@
-// module.exports = {
-//   locals: function(options) {
-//     // Return custom template variables here.
-//     return {};
-//   },
-// 
-// fileMapTokens: function(options) {
-//     // Return custom tokens to be replaced in your files
-//     return {
-//       __token__: function(options){
-//         // logic to determine value goes here
-//         return 'value';
-//       }
-//     }
-//   },
-// 
-//  Should probably never need to be override
-//
-//   filesPath: function() {
-//     return path.join(this.path, 'files');
-//   },
-// 
-// beforeInstall: function(options) {},
-//   afterInstall: function(options) {},
-// };
+module.exports = {
```

Comment:
All of the options here are controlling the presence of a chunk of the template.  For the sake of clarity and until something more apt comes along, lets this a "Feature"

Features are going to require some presence in projectSettings.  Either a boolean flag: show it or don't .  The examples here are all booleans.  More complicated objects can be anything that json represent.  If you called command like `bp gen blueprint --command.aliases=foo,bar`  that means you are going to display that feature, and you have some data meant just for it. and it should be in `{"bp": {"blueprint": {"aliases": "foo,bar"}}}`

**bp** as the leader there is to keep the top level from colliding between data for the cli, top level defaults,  and data for individual blueprints.  Any thoughts?

Reply:
Agreed. The first stab at modifying the blueprint blueprint was just to show the new functionality working. The aliases option is an excellent one to show something more than a simple boolean. Not so good at showing a default in the rc file.

I could also imagine expanding the duck example to have more complex options. E.g. an array of types or multiple options that must be defined together (namespace/slice/types). I held off going down that route until we had an opportunity to discuss the basic implementation. I'm happy to look into that some more.

Reply:
Expanding Duck and Form blueprints to allow options is required as much for being example of how it can be done as for the usefulness of the blueprint itself.

Reply:
👍 

I'll put together an example of duck options. Once we're happy with them we can look to build the ejs template(s) behind them. I'll leave the form blueprint changes to #5.